google/cloud/storage/tests/bucket_integration_test.cc

```diff
@@ -749,6 +749,278 @@ TEST_F(BucketIntegrationTest, TestBucketIamPermissionsFailure) {
       "exceptions are disabled");
 #endif  // GOOGLE_CLOUD_CPP_HAVE_EXCEPTIONS
 }
+
+TEST_F(BucketIntegrationTest, ListAccessControlFailure) {
+  Client client;
+  std::string bucket_name = MakeRandomBucketName();
+
+  // This operation should fail because the target bucket does not exist.
+#if GOOGLE_CLOUD_CPP_HAVE_EXCEPTIONS
+  EXPECT_THROW(try { client.ListBucketAcl(bucket_name); } catch (
+      std::runtime_error const& ex) {
+    EXPECT_THAT(ex.what(), HasSubstr("Permanent error in"));
```

Comment:
Is there any way to factor a lot of this out into a helper function? Something like:

    void TestForPermanentError(std::function<void()> test_func) {
    #if GOOGLE_CLOUD_CPP_HAVE_EXCEPTIONS
      EXPECT_THROW(
          try {
            test_func();
          } catch (std::runtime_error const& ex) {
            EXPECT_THAT(ex.what(), HasSubstr("Permanent error in"));
            throw;
          },
          std::runtime_error);
    #else
      EXPECT_DEATH_IF_SUPPORTED(test_func(), 
                                                                 "exceptions are disabled");
    #endif
    }

That way, the tests would become a bit shorter:

    TEST_F(BucketIntegrationTest, TestNameHere) {
      auto failing_func = [&client, bucket_name]() {
        client.ListBucketAcl(bucket_name);
      }
      TestForPermanentError(failing_func);
    }

Not sure if something like that is doable or not, but it would clean these up quite a bit, IMO.  Thoughts?